src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -341,20 +341,20 @@ pListItem = do
                            (Plain ils:xs) -> B.fromList (Plain
                                 [Span (ident, [], []) ils] : xs)
                            _ -> B.divWith (ident, [], []) bs
-  maybe id addId (lookup "id" attr) <$>
-    pInTags "li" (pLabel' <|> block)
-  where 
-    pLabel' :: PandocMonad m => TagParser m Blocks
-    pLabel' = try $ pInTag TagsRequired "label" pInput'
-    pInput' :: PandocMonad m => TagParser m Blocks
-    pInput' = do
-          TagOpen _ attr' <- lookAhead $ pSatisfy $ matchTagOpen "input" [("type","checkbox")]
-          let attr = toStringAttr attr'
-          let isChecked = isJust $ lookup "checked" attr
-          let escapeSequence = B.Str $ if isChecked then "\9746" else "\9744"
-          _ <- pSelfClosing (== "input") (const True)
-          bs <- inline
-          return $ B.Many . Seq.singleton $ B.Plain ([escapeSequence, B.Space ] <> B.toList bs)
+  maybe id addId (lookup "id" attr) <$> 
+    pInTags "li" (blockWithPotentialCheckbox)
+  where
+    blockWithPotentialCheckbox :: PandocMonad m => TagParser m Blocks
+    blockWithPotentialCheckbox = let p = (<>) <$> parseCheckbox <*> inline
+          in B.plain . trimInlines . mconcat <$> many1 (try p <|> inline)
```

Comment:
Could this be simplified to `many1 (try parseCheckbox <|> inline)`?  (And remove `p`)
Note:  this (and the current code) does not require that the checkbox be at the beginning; is that okay? Probably?

Reply:
I don't think it can be simplified to that particular code because my current assumption is if we see a checkbox we want to see at least 1 inline after it.
`many1 (try parseCheckbox <|> inline)` would potentially parse only a checkbox

Reply:
Can't there be empty task list items?

Reply:
Decided to remove this code in favor of the old approach w/ added inline parser